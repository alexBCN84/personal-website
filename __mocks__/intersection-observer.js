/**
 * Utility function that mocks the `IntersectionObserver` API. Necessary for components that rely
 * on it, otherwise the tests will crash. Recommended to execute inside `beforeEach`.
 * @param {object} intersectionObserverMock - Parameter that is sent to the `Object.defineProperty`
 *      overwrite method. `jest.fn()` mock functions can be passed here if the goal is to not only
 *      mock the intersection observer, but its methods.
 */
export function setupIntersectionObserverMock({
    observe = () => null,
    unobserve = () => null,
    disconnect = () => null,
  } = {}) {
    class IntersectionObserver {
      observe = observe;
      unobserve = unobserve;
      disconnect = disconnect;
    }
    Object.defineProperty(
      window,
      'IntersectionObserver',
      { writable: true, configurable: true, value: IntersectionObserver }
    );
    Object.defineProperty(
      global,
      'IntersectionObserver',
      { writable: true, configurable: true, value: IntersectionObserver }
    );
  }