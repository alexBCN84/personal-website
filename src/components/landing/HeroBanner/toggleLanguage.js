export function toggleLanguage(styles, element) {
    const { fixedPosition, relativePosition } = styles;
    return function toggleLanguageButtonPosition(entries) {
        // check if section is on view and is intersecting
        if (!entries[0].isIntersecting && !entries[0].isVisible) {
            return element.style.position = relativePosition;
        } else {
            return element.style.position = fixedPosition;
        }
    }
}