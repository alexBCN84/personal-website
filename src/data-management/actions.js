export const SET_ACTIVE_LANGUAGE = 'set language';