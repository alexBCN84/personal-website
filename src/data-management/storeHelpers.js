import { graphql, useStaticQuery } from 'gatsby';
import React, { createContext } from 'react';

export const StoreContext = createContext();

export function useStore() {
    const store = React.useContext(StoreContext);
    if (!store) {
    throw new Error('cannot use `useStore` outside of a storeProvider');
    }
    return store;
}

export function useData() {
    const data = useStaticQuery(
      graphql`
        query {
          file(relativePath: { eq: "logo.png" }) {
            childImageSharp {
              fixed(width: 40, height: 40) {
                ...GatsbyImageSharpFixed
              }
            }
          }
          allDataJson {
            edges {
              node {
                ES {
                  kicker
                  headline
                  description
                  aboutChevronText
                  cvCTA {
                    text
                    href
                  }
                }
                EN {
                  kicker
                  headline
                  description
                  aboutChevronText
                  cvCTA {
                    text
                    href
                  }
                }
              }
            }
          }
        }
      `
    )

    if (!data) {
      throw new Error('cannot use `useData` without data passed to a storeProvider');
    }
    return { data };
  }