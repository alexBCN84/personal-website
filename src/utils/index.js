export function scrollTo(e, time, where) {
    const eTop = e.getBoundingClientRect().top;
    const eAmt = eTop / 100;
    let curTime = 0;
    while (curTime <= time) {
        window.setTimeout(SVS_B, curTime, eAmt, where);
        curTime += time / 100;
    }
}

export function SVS_B(eAmt, where) {
    if(where === "center" || where === "") {
        window.scrollBy(0, eAmt / 2);
    }
       
    if (where === "top") {
        window.scrollBy(0, eAmt);
    }

}