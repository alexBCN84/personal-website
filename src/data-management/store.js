import React, { createContext, useReducer } from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { SET_ACTIVE_LANGUAGE } from './actions';

const initialState = {
    languages: { active: 'EN', inactive: 'ES' }
};

const store = createContext(initialState);
const { Provider } = store;

const DataProvider = ({ children }) => {
    const [state, dispatch] = useReducer((state, action) => {
        switch (action.type) {
            case SET_ACTIVE_LANGUAGE:
                return {...state, languages: action.languages}
            default:
                throw new Error();
        }
    }, initialState)
    const data = useStaticQuery(
        graphql`
          query {
            file(relativePath: { eq: "logo.png" }) {
              childImageSharp {
                fixed(width: 40, height: 40) {
                  ...GatsbyImageSharpFixed
                }
              }
            }
            allDataJson {
              edges {
                node {
                  ES {
                    kicker
                    headline
                    description
                    aboutChevronText
                    cvCTA {
                      text
                      href
                    }
                  }
                  EN {
                    kicker
                    headline
                    description
                    aboutChevronText
                    cvCTA {
                      text
                      href
                    }
                  }
                }
              }
            }
          }
        `
      );
      const logo = data.file.childImageSharp.fixed
      const { headline, kicker, description, cvCTA, aboutChevronText: about } = data?.allDataJson.edges[0].node[
        state?.languages.active
      ];
      const heroBannerData = {logo, headline, kicker, description, cvCTA, about}
    return <Provider value={{heroBannerData, dispatch, state}}>{children}</Provider>
}

export {store, DataProvider};