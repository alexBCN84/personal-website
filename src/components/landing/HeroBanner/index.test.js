import { setupIntersectionObserverMock } from "../../../../__mocks__/intersection-observer"; 
import { HeroBanner } from '../index';
import { cleanup, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from '../../../data-management/store';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import { scrollTo } from '../../../utils/index';
import { toggleLanguage } from './toggleLanguage';

// mock the implementation of the scrollTo function
jest.mock('../../../utils/index');
jest.mock('./toggleLanguage');

expect.extend({ toMatchDiffSnapshot });

const defaultMockData = {
  heroBannerData: {
    logo: {
      base64: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAC4jAAAuIwF4pT92AAACU0lEQVQ4y22Vz0tVURDH73u+vBFYIYhERItq3x8QEQgWZYsMikDoJ5VRq2zRxnDjRpAgqk3Zs0KJeBUtoq3VIgoKKgrLhH4aRLnQRT/o3T5D35Hxdg98mTnnzHzPnJk59yZZliU2TIJyooG+uV6vV5ET4JfwFgyDjuBXjhyRrGQ6JG3oL7N/4zcYY+0q8gq4J2Ib46y3B///yZAnZfwZ7MGhKckN1heBLvBBtn3zSHEqRzLmo4g0R3IAdIOdYKHWFoCqfPp0u7I7tOm0kUCSymit9h7psOW5wy5of2tcfIXxJydBNoTIR+RwRgVpdRvJkor1DlRsYYscdsugEnK6Rlc6jdgFbqG3eM7QK9K3i2OHLQ6rck3hRD/9rAxXgb3gJmgJ7eK3SMF3cMMmE5COFVS8VWRDmh/DruaEBdW/AyZN+Wm5Caf6NXpFeFQRHrEIQLMqvEJYKfvBTBFFQr/qYiXZx0ewX1c2wrta/2opVludcsLX4L4XRPKQHE6AjWAdOBwi7AT94JvsrLWuWbOb80XwBywJRXljzy+XI7tyzftQVbfRo/k0uJ0oAhsHtdEZ24hC+MvoRr8ufbVsHmjeoXmXn/5M+bBkPwYzoNEjDhE+gdTyOKXcL9PeOPgy92RR1uuEGg5DVoDQ5N7Em5CXdLUpf2rIc3NNnatsjzYux++iPUGbI0vKbxryel4+g952+W/hcRlY9fYxX1rQwKme2qSe5UDYK/wmbgDPRfwDPFTUVX1gZ7X3HmwLZKV5rCJtCFdtB5azpyqYRf0CjKoTKkW/gL+tuhI3CufaNQAAAABJRU5ErkJggg==',
      height: 40,
      src: '/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png',
      width: 40,
      srcSet: '/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png 1x,/static/b0ddc18a1e874e013ee42035351d5925/183c2/logo.png 1.5x,/static/b0ddc18a1e874e013ee42035351d5925/829e2/logo.png 2x'
    },
    about: 'about us',
    cvCTA: {
      text: 'See my CV',
      href: 'https://alexgines.netlify.app/'
    },
    description: 'I\'ve spent most of my life between Spain, France, the United Kingdom and the United States. I\'m currently based in Berlin, where I work as a Frontend Engineer at Zalando SE',
    headline: 'I am a former translator, a teacher at heart, and a Web Developer since 2016',
    kicker: 'Welcome to my Site'
  },
  state: {
    languages: {
      active: 'EN', 
      inactive: 'ES'
    },
  },
  dispatch: jest.fn()
}

global.ownerDocument = {};
global.ownerDocument.defaultView = global;



const renderHeroBanner = (data = defaultMockData) => (
  render(
    <Provider value={data}>
      <HeroBanner/>
    </Provider>
  )
)


const spyScrollTo = jest.fn();
Object.defineProperty(global.window, 'scrollTo', { value: spyScrollTo });

describe('HeroBanner', () => {
  beforeEach(setupIntersectionObserverMock);
  afterEach(cleanup); 

  it('renders without crashing', () => {
    const { container } = renderHeroBanner();
    expect.assertions(1);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('renders UI for the different breakdowns', () => {
    it('renders desktop UI properly', () => {
      window.innerWidth = 1600;
      renderHeroBanner();
      expect.assertions(1);
      expect(screen.getByTestId('desktop')).toBeInTheDocument();
    });
  
    it('renders tablet UI properly', () => {
      window.innerWidth = 800;
      renderHeroBanner();
      expect.assertions(1);
      expect(screen.getByTestId('tablet')).toBeInTheDocument();
    });
  
    it('renders mobile UI properly', () => {
      window.innerWidth = 600;
      renderHeroBanner();
      expect.assertions(1);
      expect(screen.getByTestId('mobile')).toBeInTheDocument();
    });
  });

  it('calls dispacth function with the correct arguments when clicking language switch', () => {
    renderHeroBanner();
    const { inactive } = defaultMockData.state.languages;
    const languageSwitchBtn = screen.getByText(inactive);
    expect.assertions(3);
    expect(languageSwitchBtn).toBeInTheDocument();
    userEvent.click(languageSwitchBtn);
    const dispacthArgument = {
      type: 'set language',
      languages: {
        active: 'ES', 
        inactive: 'EN'
      }
    }
    expect(defaultMockData.dispatch.mock.calls.length).toBe(1);
    expect(defaultMockData.dispatch.mock.calls[0][0]).toStrictEqual(dispacthArgument);
  });

  it('updates the DOM when clicking language switxh button', () => { 
    const { asFragment } =  renderHeroBanner();
    const { inactive } = defaultMockData.state.languages;
    const languageSwitchBtn = screen.getByText(inactive);
    expect.assertions(1);
    userEvent.click(languageSwitchBtn);
    const firstRender = asFragment();
    expect(firstRender).toMatchDiffSnapshot(asFragment());
  });

  it('renders text in English when inactive language `ES`', () => {
    renderHeroBanner();
    const { inactive } = defaultMockData.state.languages;
    const languageSwitchBtn = screen.getByText(inactive);
    const { about, cvCTA, description, headline, kicker } = defaultMockData.heroBannerData; 
    expect.assertions(6);
    expect(languageSwitchBtn).toHaveTextContent('ES');
    expect(screen.getByRole('heading', {name: kicker})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: headline})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: description})).toBeInTheDocument();
    expect(screen.getByRole('link', {name: cvCTA.text})).toBeInTheDocument();
    expect(screen.getByText(about)).toBeInTheDocument();
  });

  it('renders text in Spanish when inactive language `EN`', () => {
    const data = {
      ...defaultMockData, 
      heroBannerData: {
        logo: defaultMockData.heroBannerData.logo,
        about: 'conócenos',
        cvCTA: {
          text: 'Ver CV',
          href: 'https://alexgines.netlify.app/'
        },
        description: 'He pasado gran parte de mi vida entre España, Francia, Reino Unido y Estados Unidos. Actualmente vivo en Berlín, donde trabajo como Ingeniero Front-End en Zalando SE',
        headline: 'Soy traductor de formación, profesor de vocación y desde 2016 desarrollador web',
        kicker: 'bienvenid@ a mi web!'
      },
      state: {
        languages: {
          active: 'ES',
          inactive: 'EN'
        }
      }
    }
    renderHeroBanner(data);
    const { inactive } = data.state.languages;
    const languageSwitchBtn = screen.getByText(inactive);
    const { about, cvCTA, description, headline, kicker } = data.heroBannerData; 
    expect.assertions(6);
    expect(languageSwitchBtn).toHaveTextContent('EN');
    expect(screen.getByRole('heading', {name: kicker})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: headline})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: description})).toBeInTheDocument();
    expect(screen.getByRole('link', {name: cvCTA.text})).toBeInTheDocument();
    expect(screen.getByText(about)).toBeInTheDocument();
  });

  it('has a link to cv', () => {
    renderHeroBanner();
    const { heroBannerData } = defaultMockData;
    expect.assertions(1);
    expect(screen.getByText(heroBannerData.cvCTA.text).closest('a')).toHaveAttribute('href', heroBannerData.cvCTA.href);
  });

  describe('it fires scrollTo for the different breakpoints', () => {
    it('fires scrollTo function on mobile', () => {
      window.innerWidth = 600;
      const { container } = renderHeroBanner();
      const aboutBtnMobile = container.querySelector('#aboutButtonMobile');
      userEvent.click(aboutBtnMobile);
      expect.assertions(1);
      expect(scrollTo).toHaveBeenCalledTimes(1);
    });

    it('fires scrollTo function on tablet', () => {
      window.innerWidth = 900;
      const { container } = renderHeroBanner();
      const aboutBtnTablet = container.querySelector('#aboutButtonTablet');
      userEvent.click(aboutBtnTablet);
      expect.assertions(1);
      expect(scrollTo).toHaveBeenCalledTimes(2);
    });

    it('fires scrollTo function on Desktop', () => {
      window.innerWidth = 1900;
      const { container } = renderHeroBanner();
      const aboutBtnDesktop = container.querySelector('#aboutButtonDesktop');
      userEvent.click(aboutBtnDesktop);
      expect.assertions(1);
      expect(scrollTo).toHaveBeenCalledTimes(3);
    });
  });

  it('fires the toggleLanguage function on load', () => {
    renderHeroBanner();
    expect.assertions(1);
    expect(toggleLanguage).toHaveBeenCalled();
  });

  it('loads language switch button with initial style of fixedPosition', () => {
    const buttonText = defaultMockData.state.languages.inactive;
    const { container } = renderHeroBanner();
    const button = screen.getByText(buttonText)
    expect(getComputedStyle(button).getPropertyValue('position')).toEqual('fixed');
  });
 }) 