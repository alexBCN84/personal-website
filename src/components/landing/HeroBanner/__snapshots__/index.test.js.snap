// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBanner renders without crashing 1`] = `
<section
  class="sc-AxjAm cbBRmT"
  data-testid="desktop"
>
  <div
    class="sc-fzozJi hhSIgG"
  >
    <div
      class="sc-AxirZ dCBnyz"
    >
      <div
        class="logo gatsby-image-wrapper"
        style="position: relative; overflow: hidden; display: inline-block; width: 40px; height: 40px;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAC4jAAAuIwF4pT92AAACU0lEQVQ4y22Vz0tVURDH73u+vBFYIYhERItq3x8QEQgWZYsMikDoJ5VRq2zRxnDjRpAgqk3Zs0KJeBUtoq3VIgoKKgrLhH4aRLnQRT/o3T5D35Hxdg98mTnnzHzPnJk59yZZliU2TIJyooG+uV6vV5ET4JfwFgyDjuBXjhyRrGQ6JG3oL7N/4zcYY+0q8gq4J2Ib46y3B///yZAnZfwZ7MGhKckN1heBLvBBtn3zSHEqRzLmo4g0R3IAdIOdYKHWFoCqfPp0u7I7tOm0kUCSymit9h7psOW5wy5of2tcfIXxJydBNoTIR+RwRgVpdRvJkor1DlRsYYscdsugEnK6Rlc6jdgFbqG3eM7QK9K3i2OHLQ6rck3hRD/9rAxXgb3gJmgJ7eK3SMF3cMMmE5COFVS8VWRDmh/DruaEBdW/AyZN+Wm5Caf6NXpFeFQRHrEIQLMqvEJYKfvBTBFFQr/qYiXZx0ewX1c2wrta/2opVludcsLX4L4XRPKQHE6AjWAdOBwi7AT94JvsrLWuWbOb80XwBywJRXljzy+XI7tyzftQVbfRo/k0uJ0oAhsHtdEZ24hC+MvoRr8ufbVsHmjeoXmXn/5M+bBkPwYzoNEjDhE+gdTyOKXcL9PeOPgy92RR1uuEGg5DVoDQ5N7Em5CXdLUpf2rIc3NNnatsjzYux++iPUGbI0vKbxryel4+g952+W/hcRlY9fYxX1rQwKme2qSe5UDYK/wmbgDPRfwDPFTUVX1gZ7X3HmwLZKV5rCJtCFdtB5azpyqYRf0CjKoTKkW/gL+tuhI3CufaNQAAAABJRU5ErkJggg=="
          style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
        />
        <picture>
          <source
            srcset="/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png 1x,/static/b0ddc18a1e874e013ee42035351d5925/183c2/logo.png 1.5x,/static/b0ddc18a1e874e013ee42035351d5925/829e2/logo.png 2x"
          />
          <img
            alt="logo"
            height="40"
            loading="lazy"
            src="/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png"
            srcset="/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png 1x,/static/b0ddc18a1e874e013ee42035351d5925/183c2/logo.png 1.5x,/static/b0ddc18a1e874e013ee42035351d5925/829e2/logo.png 2x"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            width="40"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png 1x,/static/b0ddc18a1e874e013ee42035351d5925/183c2/logo.png 1.5x,/static/b0ddc18a1e874e013ee42035351d5925/829e2/logo.png 2x" /&gt;&lt;img loading="lazy" width="40" height="40" srcset="/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png 1x,/static/b0ddc18a1e874e013ee42035351d5925/183c2/logo.png 1.5x,/static/b0ddc18a1e874e013ee42035351d5925/829e2/logo.png 2x" src="/static/b0ddc18a1e874e013ee42035351d5925/ff4be/logo.png" alt="logo" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
      <div
        class="sc-AxhCb ehpxdv"
      >
        <div
          class="sc-AxgMl hzwiGK"
        >
          Alejandro Gin√©s
        </div>
        <div
          class="sc-Axmtr blfvLt"
        >
          Web Developer
        </div>
      </div>
    </div>
    <div
      class="sc-fznWqX jqXnrH"
      id="aboutButtonDesktop"
    >
      <p
        class="sc-fzqARJ bqKifG"
      >
        about us
      </p>
      <div
        class="sc-fzoyTs buwGoh"
      />
    </div>
  </div>
  <div
    class="sc-fzpans fWuNxB"
  >
    <a
      class="sc-fznyAO eQYrUP"
      id="language-switch"
      style="position: fixed;"
    >
      ES
    </a>
    <div
      class="sc-fznZeY deXbxg"
    >
      <h2
        class="sc-fzokOt dpyQpq"
      >
        Welcome to my Site
      </h2>
      <h1
        class="sc-fzoyAV gDGNOH"
      >
        I am a former translator, a teacher at heart, and a Web Developer since 2016
      </h1>
      <h3
        class="sc-fzpjYC enOmxh"
      >
        I've spent most of my life between Spain, France, the United Kingdom and the United States. I'm currently based in Berlin, where I work as a Frontend Engineer at Zalando SE
      </h3>
      <div
        class="wrapper"
      >
        <div
          class="link_wrapper"
        >
          <a
            class="cv_button"
            href="https://alexgines.netlify.app/"
            target="blank"
          >
            See my CV
          </a>
          <div
            class="icon"
          >
            <svg
              id="Capa_1"
              style="width: 34px; height: 34px; vertical-align: middle;"
              version="1.1"
              viewBox="0 0 512 512"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <g>
                <g>
                  <path
                    d="M333.205,34.213L330.799,0L9.546,22.591l30.317,431.118l44.011-3.095V512h322.046v-35.276l50.265,5.413l46.27-429.698 L333.205,34.213z M83.874,420.519l-16.17,1.137L41.599,50.432l261.359-18.38l3.359,47.765H83.874V420.519z M375.9,481.979H113.895 V109.838H375.9V481.979z M429.552,449.073l-23.631-2.545V79.817h-69.508l-1.068-15.18l134.049,14.434L429.552,449.073z"
                    fill="#fff"
                  />
                </g>
              </g>
              <g>
                <g>
                  <rect
                    fill="#fff"
                    height="30.021"
                    width="195.796"
                    x="147.003"
                    y="296.746"
                  />
                </g>
              </g>
              <g>
                <g>
                  <rect
                    fill="#fff"
                    height="30.021"
                    width="195.796"
                    x="147.003"
                    y="350.143"
                  />
                </g>
              </g>
              <g>
                <g>
                  <rect
                    fill="#fff"
                    height="30.021"
                    width="195.796"
                    x="147.003"
                    y="403.54"
                  />
                </g>
              </g>
              <g>
                <g>
                  <path
                    d="M201.384,229.293c-13.238,0-22.154-9.186-22.154-23.64s8.915-23.64,22.154-23.64c8.511,0,15.13,4.053,20.668,11.078 l20.128-17.966c-9.321-12.157-23.911-18.912-42.418-18.912c-30.394,0-52.684,20.398-52.684,49.442s22.289,49.442,52.684,49.44 c18.508,0,33.097-6.754,42.418-18.912l-20.128-17.966C216.514,225.24,209.895,229.293,201.384,229.293z"
                    fill="#fff"
                  />
                </g>
              </g>
              <g>
                <g>
                  <polygon
                    fill="#fff"
                    points="321.075,158.373 297.975,213.624 275.415,158.373 241.102,158.373 281.088,252.934 312.429,252.934  352.414,158.373"
                  />
                </g>
              </g>
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`HeroBanner updates the DOM when clicking language switxh button 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
